src/roles/common/tasks/check-predeploy-prereq.yml

```diff
@@ -20,8 +20,19 @@
 
 - block:
 
-  - block:
+  - name: Check that image files are not missing
+    set_fact:
+      is_file_missing: "{{ image_path | default ('MISSING') == 'MISSING' or vrs_package_path | default ('MISSING')  == 'MISSING' or dkms_package_path | default ('MISSING') == 'MISSING' }}"
+  
+  - name: Delete the md5 files
+    file:
+      path: "{{ inventory_dir }}/deployment.md5.yml"
+      state: absent
+    when: is_file_missing 
+    delegate_to: localhost
 
+  - block:
```

Comment:
You will want to remove this block so that the checks remain the same for all of the tasks dealing with missing image files.